### PR TITLE
statistics: enable the new PQ

### PR DESCRIPTION
**Pull request by Rustin170506**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/55906

Problem Summary:

### What changed and how does it work?

This PR is based on https://github.com/pingcap/tidb/pull/56435

This PR enables the new PQ by default.

1. Check if the auto-analyze ratio has been changed and rebuild the queue.
2. Check if the pruning mode has been changed and rebuild the queue.
3. Handle failed jobs within a background worker to avoid requeue the failed job immediately. Also, make sure the analyze API can return the status of the execution.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by Rustin170506** [comment]
/hold

Please merge this pull request after checking out the release-8.4 branch.

File: lightning/tests/lightning_checkpoint_chunks/run.sh
Comment (Rustin170506) [inline lightning/tests/lightning_checkpoint_chunks/run.sh:118]: @Leavrth Do we have to check this value? Can we skip the waiting for the auto-analyze and check if the modified count is 5000?

File: pkg/statistics/handle/autoanalyze/internal/heap/heap.go
Comment (Rustin170506) [inline pkg/statistics/handle/autoanalyze/internal/heap/heap.go:109]: I will simplify the API in my next PR.

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2.go
Comment (Rustin170506) [inline pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2.go:1]: I will rename the file in my next PR.

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Review by qw4990 — APPROVED** [review]

File: lightning/tests/lightning_checkpoint_chunks/run.sh
Comment (Leavrth) [reply lightning/tests/lightning_checkpoint_chunks/run.sh:118]: cc @lance6716

Comment (lance6716) [reply lightning/tests/lightning_checkpoint_chunks/run.sh:118]: the purpose is to make sure lightning physical import will trigger auto analyze in the future. We can have other way to check it. sleep 130s is unacceptable.

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2.go
Comment (time-and-fate) [inline pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2.go:448]: Should we put this line after `recreateAndPushJobForTable()` since that method may fail?

Comment (Rustin170506) [reply pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2.go:448]: I believe that removing it directly is safer. If the `recreateAndPushJobForTable` fails, it indicates that we were unable to obtain the start timestamp of the txn or push the job into the queue. This situation is unlikely to occur, but if it does, it suggests that there may be an issue with the queue or the handle. Therefore, we probably shouldn't remove it to prevent unnecessary retries in the RequeueFailedJobs. Additionally, if a table has been dropped, we should also clean up the failed job.

File: pkg/statistics/handle/autoanalyze/refresher/refresher.go
Comment (time-and-fate) [inline pkg/statistics/handle/autoanalyze/refresher/refresher.go:42]: This line can be removed.

**Review by time-and-fate — APPROVED** [review]
Others LGTM.

File: lightning/tests/lightning_checkpoint_chunks/run.sh
Comment (Rustin170506) [reply lightning/tests/lightning_checkpoint_chunks/run.sh:118]: > I think it's OK if stats team has enough confident that "modified count is 5000" will trigger an auto analyze eventually.

I think we still need to verify this behavior. I will try to use a failpoint to trigger the auto-analyze check immediately.

File: pkg/statistics/handle/autoanalyze/refresher/refresher.go
Comment (Rustin170506) [reply pkg/statistics/handle/autoanalyze/refresher/refresher.go]: Updated.

**Review by lance6716 — APPROVED** [review]

**Pull request merged**
